builtin/providers/cloudinit/resource.go

```diff
@@ -0,0 +1,274 @@
+package cloudinit
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/textproto"
+	"sort"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCreate,
+		Delete: resourceDelete,
+		Exists: resourceExists,
+		Read:   resourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"part": &schema.Schema{
+				Type:     schema.TypeSet,
```

Comment:
My original implementation here was deficient - I think it turns out we can use `schema.TypeList` and then not require the ordering specification.